HGustavs/LenaSYS
# Fix for issue #12141, added a missing semicolon that fixes issue #12632 and updated install file to fix issue #12659

#12141  #12642 #12659 
Moved the diagram specific functions in showdugga.php to diagram.php, stored the variables in the url in javascript variables to be used in diagram_dugga.html for both duggaed.php and showdugga.php, moved where the diagram.js and diagram_dugga.js finds the moved functions ,updated the showfacit in  diagram_dugga.js and added so the getVariantPreview fetches the duggaed.js after the templete.js because they got functions with the same name.Added a missing semicolon in showduggaservice.php. updated the install file.
To test:
Open diagram dugga and see that it works as after the latest merge.
Log in as toddler 
Go to show tests(paper in the middle of the navbar)
Select diagram dugga by clicking the pen
Press the play button on the variant you want to preview(if there is none create one)
The preview should work as if you are a student with that variant.
Check the funcionality of the preview dugga is everything as the variant states
